libtock/crt0.c

```diff
@@ -228,37 +238,82 @@ void _start(void* app_start __attribute__((unused)),
 #endif
 }
 
+// C startup routine that configures memory for the process. This also handles
+// PIC fixups that are required for the application.
+//
+// Arguments:
+// - `app_start`: The address of where the app binary starts in flash. This does
+//   not include the TBF header or any padding before the app.
+// - `mem_start`: The starting address of the memory region assigned to this
+//   app.
 __attribute__((noreturn))
-void _c_start(uint32_t* app_start, uint32_t stacktop) {
+void _c_start(uint32_t app_start, uint32_t mem_start) {
   struct hdr* myhdr = (struct hdr*)app_start;
 
-  // fix up GOT
-  volatile uint32_t* got_start     = (uint32_t*)(myhdr->got_start + stacktop);
-  volatile uint32_t* got_sym_start = (uint32_t*)(myhdr->got_sym_start + (uint32_t)app_start);
+  // Fix up the Global Offset Table (GOT).
```

Comment:
I'm confused. What's even in the GOT for non-PIC applications? Is `got_size` just set to zero?

Reply:
No. I wouldn't have found this issue otherwise :)

Reply:
But all the fixup in the for loop only works for PIC code compiled with the 0x80000000 and 0x00000000 trick. So either `myhdr->got_size` is zero in practice, the GOT is being fixed up improperly for non-PIC code, or I'm misunderstanding something.

Reply:
Ok yeah I'm getting confused about what is in what PR. Yes, RISC-V apps have entries in the GOT. It seems like it might have something to do with global variables. No, they do not currently work in libtock-c, for exactly the reason you mentioned. The riscv-for-all PR includes two implementations for c_start, one with PIC, one without. However, that change makes the diff worse, so I didn't include it in this PR.

Reply:
Alright, I'll prod over there about this issue then.